### ISSUE TITLE
Rotation of entities in existing worlds broken

### ISSUE THREAD

**Issue by paramat** [minetest/minetest#7981]
![screenshot_20181215_080352](https://user-images.githubusercontent.com/3686677/50040680-29297300-0040-11e9-9069-213a9a85b12a.png)

^ Some boats in 0.4.16

![screenshot_20181215_080230](https://user-images.githubusercontent.com/3686677/50040684-38a8bc00-0040-11e9-8ed1-dea51c71a5b1.png)

^ Same world moved to 5.0.0-dev f0dca284b3d4fe266e88f6df8c6e3ad9d1411496

Note the boat on the right corrected itself once i attached and moved it around, however not all entities are drivable vehicles.

Issue first reported here https://github.com/minetest/minetest/issues/7927#issuecomment-447513057 but deserves a separate issue.

**Comment by pgimeno** [comment]
Is there anything like a serialization sub-version number for entity data? If so, that version would allow deciding whether to read just one float or three.

**Comment by jastevenson303** [comment]
Can't the object simply be replaced?  Or just do /clearobjects?

**Comment by CoderForTheBetter** [comment]
Yeah, this isn't the best look for an update. There might be some way to tell if an entity is using old rotation or not, will have to look into more.

**Comment by SmallJoker** [comment]
PR available: #7989 please review to fix this issue.
